Add more Maxim 5x5mm TQFN footprints

matthijskooijman opened the pull request:
This adds all Txx55 footprints from the Maxim 21-0140 datasheet that were not already added. In addition, this addds thermal vias to two existing TQFN footprints that did not have them yet.

I have some remarks about these footprints and the way they are generated. To keep discussion in one place, I've noted them in https://github.com/KiCad/kicad-footprints/pull/2008

poeschlr commented:
Could you build this on top of https://github.com/pointhi/kicad-footprint-generator/pull/422 to get the corrected behaviour for EP rounding?

chschlue commented:
Pending #422

matthijskooijman commented:
I would think it is better to wait for that PR to be merged, and then rebase this one on top of it. If not, let me know, and I can rebase it now.

chschlue commented:
Oh, I wasn't trying to get you to act, I was just rounding up references over at 422.
Sorry for the misunderstanding.

matthijskooijman commented:
@chschlue You were not, but @poeschlr's comment before you read that way :-)

chschlue commented on scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml:
```diff
@@ -38,6 +51,64 @@ TQFN-16-1EP_3x3mm_P0.5mm_EP1.23x1.23mm:
+  device_type: 'TQFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'https://pdfserv.maximintegrated.com/package_dwgs/21-0140.PDF'
```
Please add T1655-4

matthijskooijman replied:
Do you mean in the `size_source` property, e.g. like here maybe?   size_source: 'https://pdfserv.maximintegrated.com/package_dwgs/21-0136.PDF (T1633-5), https://pdfserv.maximintegrated.com/land_patterns/90-0032.PDF'

Or should this be in the `part_number` (I guess not, since this is essentially a generic footprint that just happens to be based on on manufacturer's datasheet)?

If so, it seems T1655-4 and T1655-4C have the same specs, should I specify both? Or just one, assuming I used one for the sizes and not the other?

chschlue replied:
Append it to size source.
One is probably enough, otherwise the list would get quite long for some of the footprints.

chschlue commented on scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml:
```diff
@@ -142,6 +226,70 @@ TQFN-20-1EP_5x5mm_P0.65mm_EP3.1x3.1mm:
+  device_type: 'TQFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'https://pdfserv.maximintegrated.com/package_dwgs/21-0140.PDF'
```
T2055-5

chschlue commented on scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml:
```diff
@@ -203,3 +351,362 @@ TQFN-24-1EP_4x4mm_P0.5mm_EP2.8x2.8mm_PullBack:
+  device_type: 'TQFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'https://pdfserv.maximintegrated.com/package_dwgs/21-0140.PDF'
```
T2855-4 T2855-5

chschlue commented on scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml:
```diff
@@ -203,3 +351,362 @@ TQFN-24-1EP_4x4mm_P0.5mm_EP2.8x2.8mm_PullBack:
+  device_type: 'TQFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'https://pdfserv.maximintegrated.com/package_dwgs/21-0140.PDF'
```
T2855-6 T2855-8

chschlue commented on scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml:
```diff
@@ -203,3 +351,362 @@ TQFN-24-1EP_4x4mm_P0.5mm_EP2.8x2.8mm_PullBack:
+  device_type: 'TQFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'https://pdfserv.maximintegrated.com/package_dwgs/21-0140.PDF'
```
T3255-6

chschlue commented on scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml:
```diff
@@ -203,3 +351,362 @@ TQFN-24-1EP_4x4mm_P0.5mm_EP2.8x2.8mm_PullBack:
+  device_type: 'TQFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'https://pdfserv.maximintegrated.com/package_dwgs/21-0140.PDF'
```
T3255-[3457]

chschlue commented on scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml:
```diff
@@ -203,3 +351,362 @@ TQFN-24-1EP_4x4mm_P0.5mm_EP2.8x2.8mm_PullBack:
+  device_type: 'TQFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'https://pdfserv.maximintegrated.com/package_dwgs/21-0140.PDF'
```
T3255-9

chschlue commented on scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml:
```diff
@@ -203,3 +351,362 @@ TQFN-24-1EP_4x4mm_P0.5mm_EP2.8x2.8mm_PullBack:
+  device_type: 'TQFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'https://pdfserv.maximintegrated.com/package_dwgs/21-0140.PDF'
```
T4055-1 T4055-2

chschlue commented on scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml:
```diff
@@ -203,3 +351,362 @@ TQFN-24-1EP_4x4mm_P0.5mm_EP2.8x2.8mm_PullBack:
+    maximum: 0.8
+
+  lead_width:
+    minimum: 0.20
```
0.15

chschlue commented on scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml:
```diff
@@ -203,3 +351,362 @@ TQFN-24-1EP_4x4mm_P0.5mm_EP2.8x2.8mm_PullBack:
+
+  lead_width:
+    minimum: 0.20
+    nominal: 0.25
```
0.20

chschlue commented on scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/tqfn.yaml:
```diff
@@ -203,3 +351,362 @@ TQFN-24-1EP_4x4mm_P0.5mm_EP2.8x2.8mm_PullBack:
+  lead_width:
+    minimum: 0.20
+    nominal: 0.25
+    maximum: 0.30
```
0.25

matthijskooijman commented:
@chschlue Thanks for your detailed review. I've fixed all of the issues you pointed out. For that, I added two new commits to fix the size_sources of existing footprints (1da71062aef1725fe7252264495feda1bda04cbf and 57f28f5658ad6c58850c634f62c7d9fbf21d89ee) and added a fixup commit for the new footprints I added for easy review (6eaa972cbbfbf3f9b2bb1f06848b8845526dd977). 

Let me know if that looks ok, then I can squash the fixup commit in the right place before this PR can be merged.

I've also rebased on top of master, so this should now include 422. I did not see any changes in radius to the existing TQFN footprints, but I suspect these were not regenerated after increasing the radius which 422 reverts (IIUC).

I've also updated the kicad-footprints PR with the newly generated footprints.

matthijskooijman commented:
> Looks like the FP update does not include 6eaa972 yet

I think it does? I see the added numbers here: https://github.com/KiCad/kicad-footprints/blob/fb28675179018157c61134106a65c04e704bccb7/Package_DFN_QFN.pretty/TQFN-40-1EP_5x5mm_P0.4mm_EP3.5x3.5mm.kicad_mod#L2
And the pads seem to correctly be 0.2mm wide now: https://github.com/KiCad/kicad-footprints/blob/fb28675179018157c61134106a65c04e704bccb7/Package_DFN_QFN.pretty/TQFN-40-1EP_5x5mm_P0.4mm_EP3.5x3.5mm.kicad_mod#L34

Or what are you missing exactly?

I also just squashed the fixup commit, so the PR should be mergeable as-is now.

chschlue commented:
You're right. I probably looked at the wrong commit range or something.
Looks good.

chschlue commented:
FPs merged

matthijskooijman commented:
Awesome!
